ddp-workspace/projects/ddp-sdk/src/lib/utility/RegExpHelper.ts

```diff
@@ -0,0 +1,50 @@
+export class RegExpHelper {
+
+    /* The method returns a RegExp:
+    *  if text includes symbols to substitute, they match with any of `substitutableSymbols` or `replaceValue`(a space by default),
+    *  (case/space insensitive)
+    *  e.g.
+    *  regExp = createSubstitutableSymbolsRegExp('asd / zxc', ['-', '/'], ' ');
+    *  regExp.test('asd zxc') = true;
+    *  regExp.test('asd / zxc') = true;
+    *  regExp.test('asd -    zxc') = true;
+    */
+    static createSubstitutableSymbolsRegExp(text: string, substitutableSymbols: string[] = [], replaceValue = ' '): RegExp {
+        const anyAmountOfSpacesRegExp = '\\s*';
+        if (substitutableSymbols.includes(text) || !substitutableSymbols.some(s => text.includes(s))) {
+            // return same text but space/case insensitive
+            return new RegExp(
+                RegExpHelper.replaceEmptySpaces(
+                    RegExpHelper.escapeRegExp(text),
+                    anyAmountOfSpacesRegExp
+                ),
+                'gi'
+            );
+        }
+
+        const escapedSubstitutableSymbols = substitutableSymbols.map(symbol => RegExpHelper.escapeRegExp(symbol));
+        const substitutableSymbolsRegExp = new RegExp(escapedSubstitutableSymbols.join('|'), 'g');
+
+        const pattern = RegExpHelper.escapeRegExp(
+                RegExpHelper.replaceEmptySpaces(text, ''),
+                // have issues with incorrect handling it in text, escaping substitutableSymbols twice due to the regex creating.
+                // Let's avoid escaping them here.
+                substitutableSymbols
+            )
+            .replace(
+                substitutableSymbolsRegExp,
+                [anyAmountOfSpacesRegExp, '[', ...escapedSubstitutableSymbols, replaceValue, ']', anyAmountOfSpacesRegExp].join('')
```

Comment:
this line might need a comment, or described test cases

Reply:
Added both.